sdk/test/trace/tracer_test.cc

```diff
@@ -81,17 +90,29 @@ TEST(Tracer, ToMockSpanExporter)
 
 TEST(Tracer, StartSpan)
```

Comment:
Can we please have two separate tests for this, like `StartSpanSampleOn` and `StartSpanSampleOff`?